appsmithorg/appsmith
# Google Recaptcha integration for button and form button widgets.

## Description
> Google Recaptcha can be configured using `Google Recaptcha Key` inside a Button and Form Button Widget.
> Once the Recaptcha key is configured in property pane, the button onClick will generate a token.
> This token can be accessed using `{{Button.recaptchaToken}}`.
> The onClick action will be called after the token is calculated. 
> There is a an error message shown if the token was not generated.

## Type of change
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## How Has This Been Tested?
> Google recaptcha v3 key needs to be generated using [this link](https://www.google.com/recaptcha/admin/create)

- Configure the google Recaptcha key in button widget and check if a new `{{Button.recaptchaToken}}` is generated after clicking.
- Configure the `onClick` action for the button. Check if the onClick action works with the Google ReCaptcha Key.
- Check the `onClick` action and `recaptchaToken` for both valid and invalid Recaptcha Keys.
- Do the above for Form Button as well.

## Checklist:
